### PR TITLE
Provider/Azure: Enrich the unit tests for azure_vmss.go.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
/kind cleanup

**What this PR does / why we need it**:
Enrich the unit tests for azure_vmss.go. The test coverage increases from 19.7% to 82.9%.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go [hunk 1]
```diff
@@ -712,7 +712,7 @@ func (ss *scaleSet) GetPrimaryInterface(nodeName string) (network.Interface, err
 		exists, realErr := checkResourceExistsFromError(rerr)
 		if realErr != nil {
 			klog.Errorf("error: ss.GetPrimaryInterface(%s), ss.GetVirtualMachineScaleSetNetworkInterface.Get(%s, %s, %s), err=%v", nodeName, resourceGroup, ssName, nicName, realErr)
-			return network.Interface{}, err
+			return network.Interface{}, realErr.Error()
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:715]: thanks for fixing the bug here

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go [hunk 2]
```diff
@@ -75,12 +79,18 @@ func buildTestVirtualMachineEnv(ss *Cloud, scaleSetName, zone string, faultDomai
...
+				},
 			},
 		}
 		ipConfigurations := []compute.VirtualMachineScaleSetIPConfiguration{
 			{
 				Name: to.StringPtr("ipconfig1"),
-				VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{},
+				VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
+					Primary:                         to.BoolPtr(true),
+					LoadBalancerBackendAddressPools: &[]compute.SubResource{{ID: to.StringPtr("/subscriptions/sub/resourceGroups/rg/providers/Microsoft.Network/loadBalancers/lb/backendAddressPools/backendpool-0")}},
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go:92]: nit: define a const for LoadBalancerBackendAddressPoolId

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go [hunk 3]
```diff
@@ -106,12 +116,16 @@ func buildTestVirtualMachineEnv(ss *Cloud, scaleSetName, zone string, faultDomai
 				},
 				InstanceView: &compute.VirtualMachineScaleSetVMInstanceView{
 					PlatformFaultDomain: &faultDomain,
+					Statuses: &[]compute.InstanceViewStatus{
+						{Code: to.StringPtr("PowerState/Running")},
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go:120]: same here, define a const for power state

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve
/retest

**Comment by saschagrunert** [comment]
Hey @nilo19, may I ask here too to change the PR release notes block to just `None`?

**Pull request merged**